dev/travis/before_install.sh

```diff
@@ -23,4 +23,56 @@ phpenv rehash;
 test -n "$GITHUB_TOKEN" && composer config github-oauth.github.com "$GITHUB_TOKEN" || true
 
 composer config --global http-basic.repo.magento.com "$MAGENTO_USERNAME" "$MAGENTO_PASSWORD"
-git clone https://github.com/magento/magento2
+# TODO: we should specify a particular SHA/branch here
+git clone --depth 1 https://github.com/magento/magento2
+
+if [[ ${TEST_SUITE} = "functional" ]]; then
+    echo 'Creating magento2 database'
+    mysql -uroot -e 'CREATE DATABASE magento2;'
+    # Install apache
+    sudo apt-get update
+    sudo apt-get install apache2 libapache2-mod-fastcgi
+    if [ ${TRAVIS_PHP_VERSION:0:1} == "7" ]; then
+        sudo cp ${TRAVIS_BUILD_DIR}/magento2/dev/travis/config/www.conf ~/.phpenv/versions/$(phpenv version-name)/etc/php-fpm.d/
+    fi
+
+    # Enable php-fpm
+    sudo cp ~/.phpenv/versions/$(phpenv version-name)/etc/php-fpm.conf.default ~/.phpenv/versions/$(phpenv version-name)/etc/php-fpm.conf
+    sudo a2enmod rewrite actions fastcgi alias ssl
+    echo "cgi.fix_pathinfo = 1" >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini
+    ~/.phpenv/versions/$(phpenv version-name)/sbin/php-fpm
+
+    # Create key and self-signed SSL certificate
+    sudo openssl req -new -newkey rsa:4096 -days 365 -nodes -x509 \
+        -subj "/C=US/ST=Denial/L=Springfield/O=Dis/CN=magento2.travis" \
+        -keyout "${MAGENTO_HOST_NAME}.key" -out "${MAGENTO_HOST_NAME}.cert"
+
+    # Configure apache virtual hosts
+    sudo cp -f ${TRAVIS_BUILD_DIR}/dev/travis/config/apache_virtual_host /etc/apache2/sites-available/000-default.conf
+    sudo sed -e "s?%MAGENTO_ROOT%?${TRAVIS_BUILD_DIR}/magento2?g" --in-place /etc/apache2/sites-available/000-default.conf
+    sudo sed -e "s?%MAGENTO_HOST_NAME%?${MAGENTO_HOST_NAME}?g" --in-place /etc/apache2/sites-available/000-default.conf
+    sudo sed -e "s?%TRAVIS_BUILD_DIR%?${TRAVIS_BUILD_DIR}?g" --in-place /etc/apache2/sites-available/000-default.conf
+
+    sudo usermod -a -G www-data travis
+    sudo usermod -a -G travis www-data
+
+    phpenv config-rm xdebug.ini
+    sudo cat /etc/apache2/sites-enabled/000-default.conf
+    sudo service apache2 restart
+
+    # Install chromedriver to power chrome
+    if [ ! -d "${HOME}/drivers" ]; then
+        mkdir -p "${HOME}/drivers"
+    fi
+    if [ ! -f "${HOME}/drivers/chromedriver" ]; then
+        pushd "${HOME}/drivers"
+        # TODO: factor out the version to travis env var
+        wget http://chromedriver.storage.googleapis.com/78.0.3904.70/chromedriver_linux64.zip
```

Comment:
We are using the latest stable chrome. Is there a dynamic link to the latest stable chromedriver?